language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -211,46 +230,46 @@ private static void setItemInfo(String line, int startPosition, TextDocumentServ
                 break;
             }
             char c = line.charAt(counter);
-            if (!(Character.isLetterOrDigit(c)
-                    || "_".equals(Character.toString(c))) || TERMINAL_CHARACTERS.contains(Character.toString(c))) {
+            if (!(Character.isLetterOrDigit(c) || "_".equals(Character.toString(c)))
+                    || TERMINAL_CHARACTERS.contains(Character.toString(c))) {
                 callableItemName = line.substring(counter + 1, startPosition + 1);
                 delimiter = String.valueOf(line.charAt(counter));
+                captureIdentifierAgainst(line, counter, signatureContext);
                 break;
             }
             counter--;
         }
         signatureContext.put(SignatureKeys.CALLABLE_ITEM_NAME, callableItemName);
         signatureContext.put(SignatureKeys.ITEM_DELIMITER, delimiter);
     }
-    
-    public static void captureIdentifierAgainst(String line, int startPosition,
+
+    /**
+     * Capture the identifier against (s.contains() here s is the identifier against).
+     * @param line              Current line being evaluated
+     * @param startPosition     Evaluation start position
+     * @param signatureContext  Signature help context
+     */
+    private static void captureIdentifierAgainst(String line, int startPosition,
                                                  TextDocumentServiceContext signatureContext) {
         int counter = startPosition;
         String identifier = "";
-        while (true) {
-            if (counter < 0) {
-                break;
-            }
-            char c = line.charAt(counter);
-            if (TERMINAL_CHARACTERS.contains(Character.toString(c))) {
-                identifier = line.substring(counter + 1, startPosition + 1);
-                break;
-            }
+        if (".".equals(Character.toString(line.charAt(counter)))
+                || ":".equals(Character.toString(line.charAt(counter)))) {
             counter--;
+            while (true) {
+                if (counter < 0) {
+                    break;
+                }
+                char c = line.charAt(counter);
+                if (TERMINAL_CHARACTERS.contains(Character.toString(c)) || Character.toString(c).equals(" ")
+                        || Character.toString(c).equals("\n")) {
```

Comment:
should we add a "\n \r" to support the windows? please check this.

Reply:
Here \r\n is also captured via the \n and trim() used for removing trailing and preceding spaces